### DISCUSSION TITLE
What would be the best way to implement a custom accent color?

### DISCUSSION THREAD

**Discussion by jonschmidt** [radix-ui/themes#206] [Q&A] [answered]
Our designer is not crazy about the idea of choosing from a pre-defined palette, and we already have an accent color that we use in every other medium outside of our website. Is it reasonable to be able to implement our own accent color, or is this discouraged for some reason? I was thinking we might be able to do this by choosing a pre-defined accent color that closes matches ours, and then overriding some tokens.

**Comment by vladmoroz** [comment]
Hey, this is a great request. We actually plan to release a tool that would allow you to easily create branded color palettes that would seamlessly slot into how Radix Colors works. The public release will be after the holidays.

For now yep, using the closest accent color and overriding the key tokens is the best way to go about it.

> we already have an accent color that we use in every other medium outside of our website

Could you share what that looks like? If you are down to try a preview version of the tool, I’ll post it within a couple days here.

**Reply by jonschmidt** [reply]
Hi @vladmoroz . Somehow I missed your request at the bottom. Not sure if this helps, but this is our primary color, and the variants
![image](https://github.com/radix-ui/themes/assets/513206/2edeabdc-f922-49fe-ab2a-4572b610940b)

I guess the primary accent color would be #79DBC0. I would love to test out your tool!

**Reply by jonschmidt** [reply]
I found this tool which does something similar which I am going to use for now, but as this has 11 colors vs 12 to radix, curious what the differences are https://uicolors.app/create

**Reply by backuardo** [reply]
@vladmoroz I'd love to try out the preview of the branded color palette tool!

**Reply by dclark27** [reply]
@vladmoroz would also be down to try a preview of the tool! we are going through a transition from shadcn to radix-themes because of the awesome figma resource you all have. the only piece missing is allowing us to define a palette to match our brand guidelines.

**Reply by jonschmidt** [reply]
Going to humbly ping this in the hopes that the tool sees the light of day in the future. We are very grateful for work on this library in general, and the time it saves us. Being able to use custom colors would boost this tremendously, as coming up with a suitable palette has not proved that easy. thanks 🙏

**Reply by vladmoroz** [reply]
> Going to humbly ping this in the hopes that the tool sees the light of day in the future. We are very grateful for work on this library in general, and the time it saves us. Being able to use custom colors would boost this tremendously, as coming up with a suitable palette has not proved that easy. thanks 🙏

Good timing, we just released this! 
https://twitter.com/radix_ui/status/1771587676019269662

**Comment by vladmoroz** [comment, marked as answer]
It’s out 🫡
https://twitter.com/radix_ui/status/1771587676019269662

Let us know what you think

**Reply by jonschmidt** [reply]
First impressions are that I really like the way it showcases the changes as you make them. However I am running into an immediate problem as I do not know what to pick for gray + surface. Most of our surfaces are flat white, so my instinct is to go with that, however gray, I have no idea. 
As I mentioned above, our brand color is #79DBC0. The Closest radix color seems to be mint by my calculations. Mint uses "sage" as the gray according to the docs, but I am unsure what that maps to. Sage-9 is #868e8b according to the radix css, but selecting that gets me to this unusable palette.
So I guess some more guidance on how to pick a good gray would be a good things to know

![image](https://github.com/radix-ui/themes/assets/513206/7286d4f5-836d-4eff-a0a8-03df59179365)

**Reply by vladmoroz** [reply]
I think you want to switch from dark mode to light mode first, you seem to be using an #FFF background for dark mode palette

> As I mentioned above, our brand color is #79DBC0. The Closest radix color seems to be mint by my calculations. Mint uses "sage" as the gray according to the docs, but I am unsure what that maps to. Sage-9 is #868e8b according to the radix css, but selecting that gets me to this unusable palette.

You don't have to use the generated gray color if you are happy with the default grays. Just copy the accent color definitions. If you use white background in light mode, no need to override that too.

**Comment by jonschmidt** [comment]
Ah of course. The dark mode vs light mode problem did not occur to me.
Thanks for the tips!

**Comment by erikvdplas** [comment]
@vladmoroz How are the 12 shades calculated from the accent / gray colors; is this algorithm open source? I'd like to be able to dynamically update the accent shades based on brand colors :) Thanks!

**Comment by timbo-tj** [comment]
+1, it would be awesome to be able to do, say 

```
<Theme customPalette={{
  light: {
    accent: '#3D63DD'
    gray: '#8B8D98',
    background: '#FFFFFF'
  },
  dark: {
    accent: '#3D63DD'
    gray: '#8B8D98',
    background: '#111111'
  }
}}>
...
</Theme>
````

And have it generate the colors for everything just as it does on the website.
